rclpy/src/rclpy/_rclpy.c

```diff
@@ -3545,6 +3545,28 @@ rclpy_get_service_names_and_types(PyObject * Py_UNUSED(self), PyObject * args)
   return pyservice_names_and_types;
 }
 
+/// Fill a given rmw_time_t with python Duration info, if possible.
+/**
+  * \param[in] pyobject Python Object that should be a Duration, error checking is done
+  * \param[out] out_time Valid destination for time data
+  * \return Whether or not conversion succeeded.
+  */
+bool
+_convert_py_duration_to_rmw_time(PyObject * pyobject, rmw_time_t * out_time)
+{
+  rcl_duration_t * duration = (rcl_duration_t *)PyCapsule_GetPointer(pyobject, "rcl_duration_t");
+  if (!duration) {
+    PyErr_Format(PyExc_TypeError,
```

Comment:
If `duration` is `NULL` then `GetPointer` already raised `ValueError`. If that exception type is fine then it's enough to return here, but if you'd like to change the type of exception set then call `PyErr_Clear()` first.

Reply:
cd6e8a5

Reply:
Ah - thanks for the heads up on these, now I will know next time. Removed the redundant error messaging.